### PR TITLE
py-pint: new versions 0.21, 0.22

### PR DESCRIPTION
**Pull request by sethrj**
Only dependency change is a requirement for python 3.9 in the newest version.

### CODE REVIEW

**Review by manuelakuhn — COMMENTED** [review]
Requires `py-setuptools@61:` as build dependencies since version `0.21`. (see [pyproject.toml#L66](https://github.com/hgrecco/pint/blob/0.21.1/pyproject.toml#L66))

While you are at it, could you also make the order of `when` and `type` consistent for this package? We prefer `when` before `type`.

**Comment by sethrj** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-pint/package.py [hunk 1]
```diff
@@ -27,11 +29,13 @@ class PyPint(PythonPackage):
...
+    depends_on("py-setuptools@61:", when="@0.21:", type="build")
+    depends_on("py-setuptools@41:", when="@0.16:0.20", type="build")
     depends_on("py-setuptools@41:", when="@0.11:0.15", type=("build", "run"))
     depends_on("py-setuptools", when="@:0.10", type=("build", "run"))
     depends_on("py-setuptools-scm@3.4.3:+toml", when="@0.11:", type="build")
     depends_on("py-setuptools-scm", when="@0.10", type="build")
-    depends_on("py-packaging", type=("build", "run"), when="@0.13:18")
-    depends_on("py-importlib-metadata", type=("build", "run"), when="@0.13:18 ^python@:3.7")
+    depends_on("py-packaging", when="@0.13:18", type=("build", "run"))
+    depends_on("py-importlib-metadata", when="@0.13:18 ^python@:3.7", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pint/package.py:42]: Missing new dep on typing-extensions

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pint/package.py (hunk 1 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/py-pint/package.py:42]: Thanks, I missed that one. Perhaps someday `spack checksum` will read the `.toml` files to generate dependencies? 😬

**Comment by sethrj** [comment]
Ping @adamjstewart

File: var/spack/repos/builtin/packages/py-pint/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pint/package.py:42]: And perhaps someday everyone will use `.toml` files 😆

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**